Assets/BossRoom/Scripts/Server/Game/Action/LaunchProjectileAction.cs

```diff
@@ -59,7 +59,8 @@ protected void LaunchProjectile()
                 m_Launched = true;
 
                 var projectileInfo = GetProjectileInfo();
-                GameObject projectile = Object.Instantiate(projectileInfo.ProjectilePrefab);
+                NetworkObject networkProjectile = NetworkObjectPool.Singleton.GetNetworkObject(projectileInfo.ProjectilePrefab);
```

Comment:
```suggestion
                // Uses pooling server side for projectils. PooledPrefabInstanceHandler will handle client side pooling automatically
                NetworkObject networkProjectile = NetworkObjectPool.Singleton.GetNetworkObject(projectileInfo.ProjectilePrefab);
```
adding comment to explain the magic